Backend/BlockChangesService.cs

```diff
@@ -36,6 +36,15 @@ public Task<List<BlockChangeRequest>> FindAllStudentRequests(Guid studentId)
             return _blockChangesCollection.Find(x => x.StudentId == studentId).ToListAsync();
         }
 
+        public async Task CancelExchangeRequest(BlockChangeRequest request)
+        {
+            await _blockChangesCollection.DeleteOneAsync(
+                x => (x.StudentId == request.StudentId &&
+                      x.BlockFrom.Equals(request.BlockFrom) &&
+                      x.BlockTo.Equals(request.BlockTo) &&
+                      x.Status == request.Status));
+        }
+
```

Comment:
Do we enable to cancel exchange in any status?
Do you deal somehow with possibility:
1. status is waiting
2. its student ask for cancel
3. other student select it to change (and we will do it) => status will change to changed
4. this code will delete it..

It would be good to log such information to know if we will have some problem later (to log file)

Reply:
I have made some changes... now it is possible to cancel only an exchange in waiting status. 
Point 3 and 4 - this code is for deleting of the exchange by user after clicking on the trash bin icon in the left panel... Not to be called after a sucessfull exchange

Reply:
I understand. I mean parallel execution. One student is deleting and second accepting the exchange at the same time... You should use probably som critical section... Or we can let it be, but there could be some errors in the future,,,